sycl/doc/extensions/proposed/sycl_ext_oneapi_graph_fusion.asciidoc

```diff
@@ -0,0 +1,777 @@
+`annotated_ptr`, the properties should be inspected by an implementation when
+the `annotated_ptr` is captured by a kernel lambda or otherwise passed as an
+argument to a kernel function. Implementations are not required to track
+internalization-related information from other USM pointers that may be used by
+a kernel, such as those stored inside of structs or other data structures.
+
+If the implementation's fusion compiler is not able to guarantee write-back of
+the final result after internalization, values stored to an internalized
+buffer/memory must not be used by any other kernel not part of the graph, as
+the data becomes unavailable to consumers through internalization. This is
+knowledge that the compiler cannot deduce. Instead, the fact that the values
+stored to an internalized buffer/memory are not used outside the fused kernel
+must be provided by the user by applying the `fusion_internal_memory` property
+to the buffer or allocated device memory as described above.
+
+The type of memory that can be used for internalization depends on the memory
+access pattern of the fuses kernel. Depending on the access pattern, the buffer
+or allocated device memory can be classified as:
+* _Privately internalizable_: If not a single element of the buffer/memory is to
+  be accessed by more than one work-item;
+* _Locally internalizable_: If not a single element of the buffer/memory is to
+  be accessed by work items of different work groups.
+
+If the implementation's fusion compiler is not able to deduce the access
+pattern, suitable information must be provided by the user. To this end,
+specializations of the `access_scope` property template defined in this
+proposal can be used to inform the fusion compiler about the access pattern of
+the kernels involved in fusion.
+
+If an `annotated_ptr` is created with any of the properties relating to
+internalization and captured by a kernel lambda or otherwise passed as an
+argument to a kernel function participating in fusion, the underlying memory
+must only be accessed via pointers that are also captured or passed as kernel
+argument. Access to the underlying memory via a different pointer, such as
+pointers stored inside of structs or other data structures results in undefined
+behavior.
+
+As already stated above, it depends on the implementation's capabilities which
+properties need to be applied to a buffer or allocated device memory to enable
+dataflow internalization. Implementations should document the necessary
+properties required to enable internalization in implementation documentation.
+
+All internalization-related properties are only _descriptive_, so it is not an
+error if an implementation is unable to perform internalization based on the
+specified properties. Implementations can provide a diagnostic message in case
+the set of specified properties are not sufficient to perform internalization,
+but are not required to do so.
```

Comment:
Unable or decides not to for other reasons, e.g., I'm thinking of a smart implementation capable of detecting private internalization would lead to spilling, so no internalization is performed.

Reply:
Done.